gcloud-java-dns/README.md

```diff
@@ -0,0 +1,324 @@
+functionality over `ZoneInfo`. In this code snippet, we create a new zone in which we intend to
+manage DNS record for domain `someexampledomain.com.`
+
+Add the following imports at the top of your file:
+
+```java
+import com.google.gcloud.storage.Zone;
+import com.google.gcloud.storage.ZoneInfo;
+```
+
+Then add the following code to create a zone.
+
+*Important: Zone names within each project have to be unique. If you choose a zone name that already
+exists, you'll get a helpful error message telling you to choose another name. In the code below,
+replace "my_unique_zone" with a unique zone name. See more about naming rules [here](https://cloud.google.com/dns/api/v1/managedZones).*
+
+*Important: The service may require that you verify ownership of the domain name that you are
+creating a zone for. Hence, we recommend that you do so beforehand. You can verify ownership of
+a domain name [here](https://www.google.com/webmasters/verification/home). Note that `gcloud-java`
+works with fully qualified domain names which must end with a period.*
+
+```java
+// Create a zone metadata
+String zoneName = "my_unique_zone"; // Change this zone name which is unique within your project
+String domainName = "someexampledomain.com."; // Change this to a domain which you own
+String description = "This is a gcloud-java-dns sample zone.";
+ZoneInfo zoneInfo = ZoneInfo.of(zoneName, domainName, description);
+
+// Create zone in Google Cloud DNS
+Zone zone = dns.create(zoneInfo);
+```
+
+You now have an empty zone hosted in Google Cloud DNS which is ready to be populated with DNS
+records for domain name `someexampledomain.com.` Upon creating the zone, the cloud service
+automatically assigned a set of DNS servers to this zone which will be hosting the records, and
+created the default required SOA and NS records for the domain inside this zone. You can read the
+list of assigned DNS servers using accessor method `zone.nameServers()` and for instance print them
+on the screen as follows. You will need to add `import com.google.gcloud.dns.DnsRecord` to your
+source code.
+
+```java
+// Print assigned name servers
+List<DnsRecord> records = zone.nameServers();
+for(DnsRecord record : records) {
+  System.out.println(record)
+}
+```
+
+You can now instruct your domain registrar to [update the domain name servers] (https://cloud.google.com/dns/update-name-servers)
+to these. As soon as this happens and the change propagates through cached values in DNS resolvers,
+all the DNS queries will be directed and answered by the Google Cloud DNS service.
+
+#### Creating DNS Records
+Now that we have an zone, we should add some DNS records. The DNS records held within zones are
+modified by so-called "change requests". In this example, we create and apply a change request to
+our zone that creates a DNS record of type A and points URL www.someexampledomain.com to
+IP address 12.13.14.15. Start by adding `import com.google.gcloud.dns.ChangeRequest` and proceed
```

Comment:
I also suggest putting this on another line.  However, if you prefer leaving imports in line, I'd suggest standardizing that over all sections in this README.